src/array/cpu/spmat_op_impl_csr.cc

```diff
@@ -508,16 +509,21 @@ CSRMatrix CSRSliceMatrix(
   std::vector<IdType> sub_data;
   sub_indptr.resize(new_nrows + 1, 0);
   const IdType kInvalidId = new_ncols + 1;
-  for (int64_t i = 0; i < new_nrows; ++i) {
+  for (int64_t i = 0; i < rows->shape[0]; ++i) {
```

Comment:
As `num_nrows` could be as large as original one if `relabel_nodes=false`, we'd better accelerate slicing via re-design the algorithm or run in parallel with `parallel_for`.